version: 1.21.8

MukjepScarlet opened the pull request:
Reverts https://github.com/CCBlueX/LiquidBounce/pull/5652

Closes #6064 

TODO:

- [x] theme shader background (works on OSX now)
- [x] weird holding item image position (Nametags) on test server (normal on OSX?)
- [x] 1.21.8 protocol, holding shield, rightclick -> shield item is gone in hotbar and nametags (normal on OSX?)
- [x] HUD items blink. especially compass (randomly happens then randomly gone???)
- [x] ItemImageAtlas broken
- [x] further optimization
- [x] 2D render migration (rewrite)
- [x] sneak packet migration
- [x] ESP 2D mode health is incorrect
- [x] DrawContext with LINES performs weird
- [x] CustomAmbience Fog
- [x] HUD hotbar tweaks don't hide vanilla exp bar

Unfixed:
- CustomAmbience Light

MukjepScarlet committed:
let's begin

MukjepScarlet committed:
fix: MinecraftFramebuffer glId

MukjepScarlet committed:
migration: ESP 2D mode

TODO: the health is incorrect

MukjepScarlet committed:
migration: FontRenderer -> Nametags/Debug

MukjepScarlet committed:
remove: ScaffoldEagle mode

MukjepScarlet committed:
chore: merge same render functions

MukjepScarlet committed:
MCEF dev build

MukjepScarlet committed:
remove SneakNetworkEvent

MukjepScarlet committed:
remove z-index of ItemStackListRenderer

MukjepScarlet committed:
fix BowAimbot OverlayTargetRenderer

MukjepScarlet committed:
fix ModuleBetterInventory

MukjepScarlet committed:
remove injection into deprecated code

MukjepScarlet committed:
fix ESP 2D

MukjepScarlet committed:
fix Minimap texture + scissor

MukjepScarlet committed:
fix Custom Fog (needs redesign)

MukjepScarlet commented:
CustomAmbience needs a redesign because Minecraft changes the fog renderer.
@ccetl could you do this?

MukjepScarlet committed:
1.21.8

MukjepScarlet committed:
refactor: public DrawContext.drawTriangle

MukjepScarlet committed:
refactor: simplify FontRenderer on underline and strikethrough

MukjepScarlet committed:
refactor: independent QuadGuiElementRenderState

VillainsRule commented:
@MukjepScarlet so 1.21.6 (partially) works?

MukjepScarlet committed:
fix: DrawContext index buffer mode

MukjepScarlet committed:
refactor: more cached collections for rendering

MukjepScarlet committed:
fix: Theme exp bar tweak

MukjepScarlet committed:
independent GUI element for FontRenderer and triangles

1zun4 commented:
> @MukjepScarlet so 1.21.6 (partially) works?

1.21.8 works without code changes, so @MukjepScarlet decided to skip it entirely. Considering the 1.21.5 version supports 1.21.6 as well, this makes sense.

MukjepScarlet committed:
independent GUI element for lines

MukjepScarlet committed:
improve ScreenRect performance

MukjepScarlet committed:
recycle matrix3x2f

MukjepScarlet committed:
chore: remove renderEnvironmentForGUI

MukjepScarlet committed:
chore: split Render2D/Render3D

MukjepScarlet committed:
chore: optimizations

MukjepScarlet committed:
fix(Nametag): name style dropped

MukjepScarlet committed:
slightly optimize ESP2D

MukjepScarlet committed:
migrate 1.21.5 sneak command packet

MukjepScarlet committed:
WebSocket event should be final state

MukjepScarlet committed:
Event cleanup

MukjepScarlet committed:
clearer identifier

MukjepScarlet committed:
std140 to other shader uniforms

MukjepScarlet committed:
fix BlockESP position

MukjepScarlet committed:
fix Trajectories

MukjepScarlet committed:
fix Render 3D relative pos

MukjepScarlet committed:
remove z-index in Nametags

MukjepScarlet committed:
cleanup code

MukjepScarlet committed:
reduce non-inline lambda in CodecBasedAdapter

MukjepScarlet committed:
tiny fix

localpthebest commented:
if only this was on the main repo so i could pull it easily

MukjepScarlet commented:
Just pull to my fork @localpthebest

localpthebest commented:
> Just pull to my fork @localpthebest

ye but then i would have a 3rd copy of lb and iont got no space for that :axoangry:
if only git had a way to pull forks

MukjepScarlet commented:
@localpthebest https://www.jetbrains.com/guide/java/tutorials/creating-a-project-from-github/adding-updating-remotes/

MukjepScarlet committed:
chore: MixinMutableText adjustment

localpthebest commented:
very halal thank you

localpthebest commented:
@MukjepScarlet whats currently broken?

localpthebest commented:
<img width="1053" height="81" alt="image" src="https://github.com/user-attachments/assets/ad63cc7e-0c8a-4d91-9b73-d9b2d74575bc" />
ye ok so nametags is broken again

localpthebest commented:
<img width="2558" height="1389" alt="image" src="https://github.com/user-attachments/assets/15e3a32e-e31a-465b-afd9-db795ee779da" />
and accelerated rendering

localpthebest commented:
sprint mode legit doesnt actually stop sprinting when consuming

MukjepScarlet commented:
> sprint mode legit doesnt actually stop sprinting when consuming

That's legal in vanilla.

MukjepScarlet commented:
> ye ok so nametags is broken again

It works to me. Please check weird cache or old stuffs

localpthebest commented:
> > sprint mode legit doesnt actually stop sprinting when consuming
> 
> That's legal in vanilla.

oh ye forgot that its possible in new versions mb for some reason grim flags it tho?

VillainsRule commented:
gr*m

DataM0del commented:
porting this to 1.21.10: MukjepScarlet#12 (30 TODOs)

MukjepScarlet commented:
> porting this to 1.21.10: MukjepScarlet#12 (30 TODOs)

No, we should release 1.21.8 and then start 1.21.10, so I'll just wait @1zun4 to fox rest things in 1.21.8

localpthebest commented:
> > ye ok so nametags is broken again
> 
> It works to me. Please check weird cache or old stuffs

proly some weird amd driver thing again

localpthebest commented:
> > porting this to 1.21.10: [MukjepScarlet#12](https://github.com/MukjepScarlet/LiquidBounce/pull/12) (30 TODOs)
> 
> No, we should release 1.21.8 and then start 1.21.10, so I'll just wait @1zun4 to fox rest things in 1.21.8

hear me out instead of bothering with 1.21.10 we just port to bedrock

MukjepScarlet committed:
Merge branch 'nextgen' into 1.21.6

MukjepScarlet committed:
remove unused file

MukjepScarlet committed:
general cleanup

also make sword item group support copper sword in 1.21.10

MukjepScarlet committed:
refactor: enable batch for debug module

MukjepScarlet committed:
fix minimap random NPE on quit game (world)

MukjepScarlet committed:
chore: FontRenderer addSimpleElement -> addPreparedTextElement

MukjepScarlet committed:
refactor: migrate ItemStackListRenderer rect bg to custom render 2D element

- Add outline in BackgroundChoice.Rect
- margin in BackgroundChoice.Rect is float instead of int now

MukjepScarlet committed:
fix: TaskProgressScreen text color

Now it requires alpha, so we set -1 as WHITE (#FFFFFFFF)

MukjepScarlet committed:
Merge branch 'nextgen' into 1.21.6

# Conflicts:
#	src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/nametags/ModuleNametags.kt

MukjepScarlet committed:
feat(Render2D): add outline to DrawContext.drawTriangle

DataM0del commented:
> hear me out instead of bothering with 1.21.10 we just port to bedrock

that's a stupid but decent idea LOL, you'd have exempts from like every ac... BUT you'd have to deal with Mojank code and no more BDS PDBs (you could reuse them on the client to update your funny faster) because, mojank...

DataM0del commented:
also it's bugrock not bedrock

MukjepScarlet committed:
cleanup

localpthebest commented:
> > hear me out instead of bothering with 1.21.10 we just port to bedrock
> 
> that's a stupid but decent idea LOL, you'd have exempts from like every ac... BUT you'd have to deal with Mojank code and no more BDS PDBs (you could reuse them on the client to update your funny faster) because, mojank...

i mean the smarter idea would be to make a bedrock translation layer instead it should be much easier than recoding to bedrock

MukjepScarlet committed:
Merge branch 'nextgen' into 1.21.6

MukjepScarlet committed:
refactor: ItemStackListRenderer param adjustment

MukjepScarlet committed:
Merge branch 'nextgen' into 1.21.6

MukjepScarlet committed:
add vec2f extension

MukjepScarlet committed:
refactor(render2d): cache `TextureSetup.of(GpuTextureView)`

MukjepScarlet committed:
refactor(nametag): formatter singleton

ManInMyVan commented:
> > > sprint mode legit doesnt actually stop sprinting when consuming
> > 
> > That's legal in vanilla.
> 
> oh ye forgot that its possible in new versions mb for some reason grim flags it tho?

only on 1.21.4 protocol

MukjepScarlet committed:
Merge branch 'refs/heads/nextgen' into 1.21.6

# Conflicts:
#	src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleBlockESP.kt
#	src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/esp/modes/Esp2DMode.kt

localpthebest commented:
> > > > sprint mode legit doesnt actually stop sprinting when consuming
> > > 
> > > 
> > > That's legal in vanilla.
> > 
> > 
> > oh ye forgot that its possible in new versions mb for some reason grim flags it tho?
> 
> only on 1.21.4 protocol

nah im on latest latest and it flage

ManInMyVan commented:
> > > > > sprint mode legit doesnt actually stop sprinting when consuming
> > > > 
> > > > 
> > > > That's legal in vanilla.
> > > 
> > > 
> > > oh ye forgot that its possible in new versions mb for some reason grim flags it tho?
> > 
> > only on 1.21.4 protocol
> 
> nah im on latest latest and it flage
> 

flags what?

localpthebest commented:
> > > > > > sprint mode legit doesnt actually stop sprinting when consuming
> > > > > 
> > > > > 
> > > > > That's legal in vanilla.
> > > > 
> > > > 
> > > > oh ye forgot that its possible in new versions mb for some reason grim flags it tho?
> > > 
> > > 
> > > only on 1.21.4 protocol
> > 
> > 
> > nah im on latest latest and it flage
> 
> flags what?

sprint

ManInMyVan commented:
> > > > > > > sprint mode legit doesnt actually stop sprinting when consuming
> > > > > > 
> > > > > > 
> > > > > > That's legal in vanilla.
> > > > > 
> > > > > 
> > > > > oh ye forgot that its possible in new versions mb for some reason grim flags it tho?
> > > > 
> > > > 
> > > > only on 1.21.4 protocol
> > > 
> > > 
> > > nah im on latest latest and it flage
> > 
> > flags what?
> 
> sprint

SprintC exempts 1.21.5 players

localpthebest commented:
> > > > > > > > sprint mode legit doesnt actually stop sprinting when consuming
> > > > > > > 
> > > > > > > 
> > > > > > > That's legal in vanilla.
> > > > > > 
> > > > > > 
> > > > > > oh ye forgot that its possible in new versions mb for some reason grim flags it tho?
> > > > > 
> > > > > 
> > > > > only on 1.21.4 protocol
> > > > 
> > > > 
> > > > nah im on latest latest and it flage
> > > 
> > > 
> > > flags what?
> > 
> > 
> > sprint
> 
> SprintC exempts 1.21.5 players

im getting some ideas...

MukjepScarlet committed:
cleanup scoreboard code

MukjepScarlet committed:
fix JSON codec can't deal with some registries

MukjepScarlet committed:
Merge branch 'refs/heads/nextgen' into 1.21.6

MukjepScarlet committed:
Merge branch 'nextgen' into 1.21.6

MukjepScarlet committed:
better cache of TextureSetup

MukjepScarlet committed:
Merge branch 'nextgen' into 1.21.6

MukjepScarlet committed:
theme bg glsl version 330 -> 410

MukjepScarlet committed:
fix: Theme shader bg (tested on Windows)

MukjepScarlet committed:
detekt

MukjepScarlet committed:
Merge branch 'refs/heads/nextgen' into 1.21.6

MukjepScarlet committed:
fix: allows joining servers that are blocked by mojang (#7273)

MukjepScarlet committed:
failed

VillainsRule commented:
so closeeeee! thanks for all of the hard work @MukjepScarlet!

localpthebest commented:
> so closeeeee! thanks for all of the hard work @MukjepScarlet!

real he doesnt get enough credit he singlehandedly carries lb

VillainsRule commented:
> > so closeeeee! thanks for all of the hard work @MukjepScarlet!
> 
> real he doesnt get enough credit he singlehandedly carries lb

izuna so useless 😤 😤 😤 need to be replaced

MukjepScarlet committed:
enabled is not correct, should be running

MukjepScarlet committed:
rewrite ItemImageAtlas

MukjepScarlet commented:
I decide to leave CustomAmbience alone.

MukjepScarlet committed:
detekt

MukjepScarlet committed:
Merge branch 'nextgen' into 1.21.6

MukjepScarlet committed:
chore: std140 ubo shortcut

MukjepScarlet committed:
chore: easier RenderPass creation

1zun4 reviewed (changes requested):
- [x] Item Image Atlas broken on Linux & Windows (?)
- [ ] The GLSL shader version is too high. GL 3.2 (which Minecraft defaults to) is specified by GLSL version 150. Previously LiquidBounce used 330 (OpenGL 3.3), however, now in this pull request we are at 410. This eliminates compatibility with all GPUs below OpenGL 4.1 and the user does not receive a proper warning.
- [x] The Connect Screen no longer shows the proxy used. The text is not being rendered for some reason.

MukjepScarlet committed:
theme bg glsl 410 -> 330

MukjepScarlet committed:
fix(ConnectScreen): text color

DataM0del commented:
the item texture renderer used to work (at least on my 1.21.10 branch) before this PR rewrote it and I merged the changes lol

MukjepScarlet commented:
> the item texture renderer

What is it

MukjepScarlet commented:
It works on MacOS but not works on Windows.

MukjepScarlet commented:
> It works on MacOS but not works on Windows.

I'm sure that current code works on OSX. No idea about other OS.

MukjepScarlet commented:
Yesterday I tested on Windows but it looks like the items are still drawn onto `mc.framebuffer` instead of custom texture.

MukjepScarlet committed:
try to fix ItemImageAtlas on Linux/Windows

MukjepScarlet committed:
RenderExtensions better Java/JS interop

MukjepScarlet committed:
refactor: directly to BufferedImage

MukjepScarlet committed:
fix ItemImageAtlas incorrect comparison

MukjepScarlet committed:
refactor(BlockESP): sync set -> concurrent sorted set

MukjepScarlet committed:
1

MukjepScarlet committed:
update sodium

MukjepScarlet committed:
Released MCEF for 1.21.6-1.21.10

MukjepScarlet committed:
Update Fabric Loom

MukjepScarlet committed:
Merge branch 'nextgen' into 1.21.6

MukjepScarlet committed:
fix ItemImageAtlas on Windows

MukjepScarlet committed:
Merge remote-tracking branch 'scarlet/1.21.6' into 1.21.6

a114mc committed:
fix AntiBlind

MukjepScarlet commented:
@DataM0del ItemImageAtlas works now

MukjepScarlet committed:
remove unused mixin

1zun4 reviewed (approved):
This version still needs some work, especially regarding rendering issues and re-adding commented-out code sections. However, these things can be completed later on.

DataM0del commented:
yay lol I'll update my 1.21.10 version

DataM0del commented:
was gonna hack on cubecraft but lol

DataM0del commented:
ok well I kind of have to redo my entire thing since it has a bunch of conflicts lol

MukjepScarlet commented:
> ok well I kind of have to redo my entire thing since it has a bunch of conflicts lol

Why? I just merged this branch to ur branch

MukjepScarlet commented:
The only thing you need to do is to cleanup the git history and rebase to main branch

DataM0del commented:
I did pull your changes, it's just that I then pulled from the nextgen branch in this repo so I had to fix a bunch of conflicts due to it being merged by squashing the changes into 1 commit lol.

DataM0del commented:
<img width="662" height="299" alt="image" src="https://github.com/user-attachments/assets/612ecc9e-62de-42bd-ac28-d7648faa1a57" />

item renderer is STILL broken, awesome

DataM0del commented:
literally just tested and it seems like latest nextgen crashes for me too aaaaaaaaaaaaaa
